lib/Backend/Lower.cpp

```diff
@@ -1863,8 +1863,20 @@ Lowerer::LowerRange(IR::Instr *instrStart, IR::Instr *instrEnd, bool defaultDoFa
                 Assert(UNREACHED);
             }
             break;
-
         case Js::OpCode::Conv_Prim:
+        {
+
+            IR::RegOpnd* src1 = instr->GetSrc1()->IsRegOpnd() ? instr->GetSrc1()->AsRegOpnd() : nullptr;
+            if (src1 && src1->m_sym->IsSingleDef() && src1->m_sym->GetInstrDef()->m_opcode == Js::OpCode::OverflowCheck)
+            {
+                //TODO : overflow checks shouldn't be CSE'ed because of side effects.
+                //DbCheck if it's safe to remove mov?
+                src1->m_sym->GetInstrDef()->m_opcode = Js::OpCode::MOV;
```

Comment:
i'm actually thinking if i should just add a case to a switch in `Lowerer::LowerRange`. It will prlly look cleaner